pkg/traceql/engine_metrics.go

```diff
@@ -338,6 +338,69 @@ func (c *CountOverTimeAggregator) Sample() float64 {
 	return c.count * c.rateMult
 }
 
+// MinOverTimeAggregator it calculates the mininum value over time. It can also
+// calculate the rate when given a multiplier.
+type MinOverTimeAggregator struct {
+	firstTime       bool
+	getSpanAttValue func(s Span) float64
+	min             float64
+	rateMult        float64
+}
+
+var _ VectorAggregator = (*MinOverTimeAggregator)(nil)
+
+func NewMinOverTimeAggregator(attr Attribute) *MinOverTimeAggregator {
+	var fn func(s Span) float64
+	switch attr {
+	case IntrinsicDurationAttribute:
+		fn = func(s Span) float64 {
+			return float64(s.DurationNanos()) / float64(time.Millisecond)
+		}
+	default:
+		fn = func(s Span) float64 {
+			return floatizeAttribute(s, attr)
+		}
+	}
+	return &MinOverTimeAggregator{
+		getSpanAttValue: fn,
+		rateMult:        1.0,
+	}
+}
+
+func (c *MinOverTimeAggregator) Observe(s Span) {
+	val := c.getSpanAttValue(s)
+	if !c.firstTime {
+		c.min = val
+		c.firstTime = true
+	} else if val < c.min {
+		c.min = val
+	}
+}
+
+func (c *MinOverTimeAggregator) Sample() float64 {
+	return c.min * c.rateMult
```

Comment:
I think we need to tweak the empty/uninitialized behavior.  In this case if we query min_over_time() and there are no matching spans, or none have the attribute, it will return zeroes.  I think not returning a sample for that time range is the most correct.  Let's chat about how to accomplish this.  This also relates to the implementation of floatizeAttribute below which doesn't indicate if the attribute was present.